src/test/java/io/wisetime/connector/utils/ActivityTimeCalculatorTest.java

```diff
@@ -35,12 +36,36 @@ void timeGroupStartTime_no_time_rows() {
 
   @Test
   void timeGroupStartTime_with_time_rows() {
-    final TimeRow row1 = fakeEntities.randomTimeRow().activityHour(2018110110);
-    final TimeRow row2 = fakeEntities.randomTimeRow().activityHour(2018110109);
-    final TimeGroup timeGroup = fakeEntities.randomTimeGroup().timeRows(ImmutableList.of(row1, row2));
+    final TimeRow row1 = fakeEntities.randomTimeRow().activityHour(2018110209).firstObservedInHour(3);
+    final TimeRow row2 = fakeEntities.randomTimeRow().activityHour(2018110209).firstObservedInHour(29);
+    final TimeGroup timeGroup = fakeEntities.randomTimeGroup().timeRows(Lists.newArrayList(row1, row2));
 
     assertThat(startTime(timeGroup))
-        .as("The start time is the segment hour of the earliest time row")
-        .contains(LocalDateTime.of(2018, 11, 1, 9, 0));
+        .as("should be the activity time of the earliest time row from the time group")
+        .contains(LocalDateTime.of(2018, 11, 2, 9, 3));
+
+    // add time row with earliest activity time by hour
+    timeGroup.getTimeRows().add(fakeEntities.randomTimeRow().activityHour(2018110208).firstObservedInHour(29));
+    assertThat(startTime(timeGroup))
+        .as("should be able to check the earliest time row by hour")
+        .contains(LocalDateTime.of(2018, 11, 2, 8, 29));
+
+    // add time row with earliest activity time by day of month
+    timeGroup.getTimeRows().add(fakeEntities.randomTimeRow().activityHour(2018110108).firstObservedInHour(29));
+    assertThat(startTime(timeGroup))
```

Comment:
Some of these seem redundant. You only really need to set up:

```
activity hour: 2018110209, first observed: 3
activity hour: 2018110209, first observed: 30
```

And assert that start time will be that of 201811020903

And another check for:

```
activity hour: 2018110209, first observed: 3
activity hour: 2018110211, first observed: 3
```

Assert that start time will be that of 201811020903

Reply:
Done

Reply:
@alvinllobrera, per your latest commit, what I meant was - we don't need to have a test for the year, month _and_ day, since the comparison is essentially a string comparison based on the concatenated year, month, day. So, two tests similar to above should be enough to cover the cases.

Reply:
Ok, got it. I've updated my PR with your suggestion. Thanks

Reply:
Thanks. I've pushed a small update. Can you review and merge if all good?